backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -427,7 +465,9 @@ public void syncState() {
         this.popped = null;
 
         for (StackFrameStep step : frame.getSteps()) {
-            if (SessionFrame.STATE_DATUM_VAL.equals(step.getType())) {
+            if (SessionFrame.STATE_DATUM_VAL.equals(step.getType()) ||
+                    SessionFrame.STATE_UNKNOWN.equals(step.getType()) &&
```

Comment:
Because the new STATE_UNKNOWN type can also possibly be a computed that we want to put into the collectedDatum table